pkg/component/component.go

```diff
@@ -414,7 +421,7 @@ func Update(client *occlient.Client, componentName string, applicationName strin
 		annotations := map[string]string{componentSourceURLAnnotation: sourceURL.String()}
 		annotations[componentSourceTypeAnnotation] = newSourceType
 
-		err = client.UpdateBuildConfig(namespacedOpenShiftObject, projectName, "", annotations)
+		err = client.UpdateBuildConfig(namespacedOpenShiftObject, projectName, "", annotations, nil)
```

Comment:
In case of local or binary component, it doesn't make sense to set env variables for BuildConfig. For those components BuildConfig just injects SupervisorD into new image, and it is not used to build application. This is why it doesn't make sense to add env variables to BC in case of local and binary components.

Also as @mik-dass mentioned, we won't be using BC for SupervisorD injection after Charlie's work is done. BuildConfig will be created only in for git components, and that BC will be regular BC that builds application.